### PR TITLE
Baseline tests for pattern 1 - Prefill

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Added unit tests and e2e tests to the `_mock-form-ae-design-patterns` application
- 3 e2e tests were addd for pattern 1 that cover the basic functionality of green, yellow, and purple tasks
- unit test suites were added for the TaskTabs and LandingPage components
- 16 total tests added, 6 e2e and 10 unit tests

- Running the tests:
  - run the watch command for the app
  - run cypress command for the desired file path glob
    - `yarn cy:run --spec src/applications/_mock-form-ae-design-patterns/tests/**`
  - run unit tests with a similar command
    - `yarn test:unit src/applications/_mock-form-ae-design-patterns/tests/**/*.unit.spec.js*`

## Related issue(s)

- https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/99

## Testing done

- Added tests for e2e and unit tests for shared components

## Screenshots

No UI changes, just adding tests

## What areas of the site does it impact?

auth patterns mock form only

## Acceptance criteria

- tests added

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by msbtterswrth — DISMISSED** [review]
There are a few warnings about prop validation that could be addressed, but otherwise nice work!

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import manifest from '../../../manifest.json';
+import mockUsers from '../../../mocks/endpoints/user';
+import mockPrefills from '../../../mocks/endpoints/in-progress-forms/mock-form-ae-design-patterns';
+
+describe('Prefill pattern - Yellow Task', () => {
```
Comment (bellepx0) [inline src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js:5]: todo: Change "Yellow Task" to "Purple Task"

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js
Comment (bellepx0) [inline src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js:1]: Question: Should we add a test for interacting with the "Edit" link?

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js:5]: Whoops! thanks for catching that.

File: src/applications/_mock-form-ae-design-patterns/App.jsx [hunk 2]
```diff
@@ -88,9 +88,9 @@ export default function App({ location, children }) {
       }
       handleEditPageDisplayTweaks(location);
 
-      // having the pollTimeout allows api calls locally
+      // having the pollTimeout present triggers api calls to be made locally and in codespaces
       if (!window?.VetsGov?.pollTimeout) {
-        window.VetsGov.pollTimeout = 5000;
+        window.VetsGov.pollTimeout = 500;
```
Comment (adamwhitlock1) [inline src/applications/_mock-form-ae-design-patterns/App.jsx:93]: this pollTImeout is what is making our codespaces slow to respond to updates to the profile info for each task. Reducing this makes the spinner go away faster

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/helpers.js
Comment (adamwhitlock1) [inline src/applications/_mock-form-ae-design-patterns/tests/e2e/helpers.js:1]: I added this to cut down on repeated code and will start using it in e2e tests to aide in the setup of each test.

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js
Comment (adamwhitlock1) [reply src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskPurple.cypress.spec.js:1]: I figured out how to mock the update api calls appropriately for the phone updates and added it to this test.

File: src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskYellow.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    cy.findByRole('link', {
+      name: /Start your health benefits update form/i,
+    }).click();
+
+    cy.wait('@mockSip');
+
+    cy.url().should('contain', '/personal-information');
+  });
+
+  it('should successfully show prefill data on the form', () => {
```
Comment (bellepx0) [inline src/applications/_mock-form-ae-design-patterns/tests/e2e/pattern1/taskYellow.cypress.spec.js:96]: suggestion: update the test name to also mention something along the lines of "...allow editing of mailing address"

**Review by bellepx0 — APPROVED** [review]
Thank you for adding these tests! I left a small suggestion, but otherwise looks good! Approved.

**Pull request merged**